src/GitVersion.Core.Tests/IntegrationTests/HotfixBranchScenarios.cs

```diff
@@ -60,7 +60,7 @@ public void CanTakeVersionFromHotfixesBranch()
 
         // create hotfix branch
         Commands.Checkout(fixture.Repository, fixture.Repository.CreateBranch("hotfixes/1.1.1"));
-        fixture.AssertFullSemver("1.1.0"); // We are still on a tagged commit
+        fixture.AssertFullSemver("1.1.1+0");
```

Comment:
Are we not still on a tagged commit here?

Reply:
Hmm very good quesiton. I think you have foun a bug. ;) The same question applies to the following scanrio:
```csharp
    [Test]
    public void __Just_A_Test__()
    {
        using var fixture = new EmptyRepositoryFixture();

        fixture.MakeATaggedCommit("1.0.0");
        fixture.BranchTo("release/1.1.0");

        fixture.AssertFullSemver("1.1.0+0");
    }
```

Why it yields to 1.1.0 and not to 1.0.0?

Reply:
I think this is one of those edge-cases that no matter which solution we pick, we are going to make some people unhappy. I'm in the "a tag should always win" camp, but I can definitely see how some people would expect `release/1.1.0` to yield `1.1.0` even though the same commit is tagged `1.0.0`. I don't think there's a right answer here, we just need to pick one way or the other, stick to it and document it well.

Reply:
I think this discussion is related to this issue here: [Same version computed on different branches](https://github.com/GitTools/GitVersion/issues/3453) #3453 

We need to introduce a new branch related property (like `take-incremetned-version`) or make the behavior dependent on the used `version-mode` option and make it configurable. The point is if you are on the hotfix branch you don't want to have properly the tagged version you would like to have the next version `1.1.0+0`. Other way around if you are on the main branch you would like to have the tagged version `1.0.0`.

![image](https://user-images.githubusercontent.com/56404113/229338350-e72c980c-3f4d-4508-8ad2-de9512137385.png)

Reply:
If we were to base this on `mode`, how would you suggest we do it?

Reply:
Hmm I think it would be not a good idea to put it on the `version-mode` option because it is very confusing and would lead to issues complaining about that the tag is not recognice correctly. What is your opinion about that?

Also if I think about the behavior of real live deplyoment scenarios (trunk-based, continues-deployment, continues-delivery and manually-deployment) this question is totally independent of that. Because it depends on the fact: Do I create a artifact after I have tagged the commit or before. Or do I take a pre-release version and just redeclare it to production version or not.

What is your opinion about that?